crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -1232,19 +1241,18 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
             return RenderCommandResult::Failure;
         };
 
-        let mut set_bind_group = |indices: &[u32]| pass.set_bind_group(I, bind_group, indices);
-        match (batch_indices.dynamic_offset, skin_index, morph_index) {
-            (None, None, None) => set_bind_group(&[]),
-            (None, Some(skin), None) => set_bind_group(&[skin.index]),
-            (None, None, Some(morph)) => set_bind_group(&[morph.index]),
-            (None, Some(skin), Some(morph)) => set_bind_group(&[skin.index, morph.index]),
-            (Some(mesh_index), None, None) => set_bind_group(&[mesh_index]),
-            (Some(mesh_index), Some(skin), None) => set_bind_group(&[mesh_index, skin.index]),
-            (Some(mesh_index), None, Some(morph)) => set_bind_group(&[mesh_index, morph.index]),
-            (Some(mesh_index), Some(skin), Some(morph)) => {
-                set_bind_group(&[mesh_index, skin.index, morph.index]);
-            }
-        };
+        let mut dynamic_offsets: SmallVec<[u32; 3]> = SmallVec::with_capacity(3);
+        if let Some(mesh_index) = batch_indices.dynamic_offset {
+            dynamic_offsets.push(mesh_index);
+        }
+        if let Some(skin_index) = skin_index {
+            dynamic_offsets.push(skin_index.index);
+        }
+        if let Some(morph_index) = morph_index {
+            dynamic_offsets.push(morph_index.index);
+        }
```

Comment:
This is very NIT and I really like the `if let Some` solution, and it's probably more efficient, but have you considered  the following:
```rust
let dynamic_offsets = [
  batch_indices.dynamic_offset,
  skin_index.map(|s| s.index),
  morph_index.map(|m| m.index),
].into_iter().flatten().collect::<SmallVec<[u32; 3]>>();
```
IMO here an [`ArrayVec`](https://docs.rs/arrayvec/latest/arrayvec/) would do the job better, since we have a dynamic size array with a known small upper bound. But I don't think it's a bevy dependency yet.